lib/components/kernel-monitor.js

```diff
@@ -1,62 +1,50 @@
 /* @flow */
 
 import React from "react";
+import ReactTable from "react-table";
+import { ReactTableDefaults } from "react-table";
 import { observer } from "mobx-react";
-import { observable } from "mobx";
 import _ from "lodash";
 import tildify from "tildify";
 
 import { KERNEL_MONITOR_URI } from "./../utils";
 
-const displayOrder = ["text/html", "text/markdown", "text/plain"];
-
 import typeof store from "../store";
 import Kernel from "../kernel";
 
-type MonitorProps = { kernel: Kernel, files: string };
-const Monitor = observer(({ kernel, files }: MonitorProps) => {
-  const destroy = () => {
-    kernel.shutdown();
-    kernel.destroy();
-  };
+const interrupt = (kernel: Kernel) => {
+  kernel.interrupt();
+};
+const shutdown = (kernel: Kernel) => {
+  kernel.shutdown();
+  kernel.destroy();
+};
+const restart = (kernel: Kernel) => {
+  kernel.restart();
+};
 
-  return (
-    <div style={{ padding: "5px 10px", display: "flex" }}>
-      <div style={{ flex: 1, whiteSpace: "nowrap" }}>{kernel.displayName}</div>
-      <div
-        style={{
-          padding: "0 10px",
-          textOverflow: "ellipsis",
-          overflow: "hidden",
-          whiteSpace: "nowrap"
-        }}
-      >
-        {files}
-      </div>
-      <div className="icon icon-trashcan" onClick={destroy} />
-    </div>
-  );
-});
+async function openFile(file: string) {
+  await atom.workspace
+    .open(file, {
+      searchAllPanes: true
+    })
+    .catch(err => {
+      atom.notifications.addError("Hydrogen", {
+        description: err
+      });
+    });
+}
```

Comment:
> I implemented "jump to file" feature just using kernel.getFilesForKernel() and atom.workspace.open(filePath) and it totally doesn't work for unsaved editors.

What I had in mind was something like 
```js
const editor = atom.workspace
  .getTextEditors()
  .find(nexEditor => nextEditor.id == relevantEditorId);

atom.workspace.open(editor);

```

The only problem is that our store is set up to use human readable/friendly filenames instead of id numbers. So you would have to parse out the id from `Unsaved Editor x` to get `relevantEditorId` for this example. I think it would be better to just move toward using ids in our store in another unrelated PR, then easily add jump to for unsaved editors.

Reply:
Phew, I implemented an easy workaround with `Unsaved Editor`'s ID to open it in [this commit](https://github.com/nteract/hydrogen/pull/1601/commits/fdaee59d469f4de428f28ad109c213bd7cc00d09) with this approach.

I also included `@TODO` comment for further implementation improvements in the future.

In anyway, now we can jump to _undestroyed_ unsaved text editors from the kernel monitor !